src/raftstore/store/worker/split_check.rs

```diff
@@ -166,20 +159,39 @@ impl<C> Runner<C> {
 
 impl<C: Sender<Msg>> Runnable<Task> for Runner<C> {
     fn run(&mut self, task: Task) {
+        let region = &task.region;
+        let region_id = region.get_id();
+
+        // Check approximate size before scanning region.
+        match util::get_region_approximate_size(&self.engine, region) {
+            Ok(size) => {
+                if size < self.region_max_size {
+                    return;
```

Comment:
need to add a metric here, like `CHECK_SPILT_COUNTER_VEC.with_label_values(&["skip"]).inc();`